qiskit/pulse/commands/parametric_pulses.py

```diff
@@ -0,0 +1,294 @@
+                        sigma=self.sigma, zero_ends=False)
+
+
+class GaussianSquare(ParametricPulse):
+    """
+    A square pulse with a Gaussian shaped risefall on either side:
+
+        risefall = (duration - width) / 2
+
+    0 <= x < risefall
+        f(x) = amp * exp( -(1/2) * (x - risefall/2)^2 / sigma^2) )
+
+    risefall <= x < risefall + width
+        f(x) = amp
+
+    risefall + width <= x < duration
+        f(x) = amp * exp( -(1/2) * (x - (risefall + width)/2)^2 / sigma^2) )
+    """
+
+    def __init__(self,
+                 duration: int,
+                 amp: complex,
+                 sigma: float,
+                 width: float):
+        """Initialize the gaussian square pulse command.
+
+        Args:
+            duration: Pulse length in terms of the the sampling period `dt`.
+            amp: The amplitude of the Gaussian and of the square pulse.
+            sigma: A measure of how wide or narrow the Gaussian risefall is; see the class
+                   docstring for more details.
+            width: The duration of the embedded square pulse.
+        """
+        validate_params(amp=amp, sigma=sigma, width=width, duration=duration)
+        self._amp = complex(amp)
+        self._sigma = sigma
+        self._width = width
+        super().__init__(duration=duration)
+
+    @property
+    def amp(self):
+        return self._amp
+
+    @property
+    def sigma(self):
+        return self._sigma
+
+    @property
+    def width(self):
+        return self._width
+
+    def get_sample_pulse(self) -> SamplePulse:
+        return gaussian_square(duration=self.duration, amp=self.amp,
+                               width=self.width, sigma=self.sigma,
+                               zero_ends=False)
+
+
+class Drag(ParametricPulse):
+    """
+    A pulse whose envelope is shaped by a drag pulse. This is so named by the technique
+    Derivative Removal by Adiabatic Gate (DRAG). It is a Gaussian pulse with a Gaussian
+    derivative component. The construction is designed to reduce the frequency spectrum
+    a normal gaussian curve near the |1>-|2> transition, reducing the chance of leakage
+    to the |2> state.
+
+        f(x) = Gaussian + 1j * beta * d/dx [Gaussian]
+             = Gaussian + 1j * beta * (-(x - duration/2) / sigma^2) [Gaussian]
+
+    where 'Gaussian' is:
+        Gaussian(x, amp, sigma) = amp * exp( -(1/2) * (x - duration/2)^2 / sigma^2) )
+
+    Ref:
+        [1] Gambetta, J. M., Motzoi, F., Merkel, S. T. & Wilhelm, F. K.
+        Analytic control methods for high-fidelity unitary operations
+        in a weakly nonlinear oscillator. Phys. Rev. A 83, 012308 (2011).
+        [2] F. Motzoi, J. M. Gambetta, P. Rebentrost, and F. K. Wilhelm
+        Phys. Rev. Lett. 103, 110501 – Published 8 September 2009
+    """
+
+    def __init__(self,
+                 duration: int,
+                 amp: complex,
+                 sigma: float,
+                 beta: int):
```

Comment:
Why is beta not a float?

Reply:
Thank you, it should be a float.

Reply:
It should still work if an int tho, which the code does. But for the type hint, yeah I think float is good.